### PR TITLE
Remove the service when uninstalling the package

### PR DESCRIPTION
**Pull request by BraulioV**
Hi team,

this PR solves the issue #75.

Regards,
Braulio.

### CODE REVIEW

File: debs/SPECS/3.7.0/wazuh-agent/debian/postrm [hunk 1]
```diff
@@ -10,12 +10,26 @@ WAZUH_TMP_DIR="${DIR}/tmp/installation_files"
...
-		delgroup ossec > /dev/null 2>&1
-	fi
+    # If the host uses systemd, disable the service and
+    # remove it from /etc/systemd/system/
+    if [ -d /run/systemd/system ] ; then
+        systemctl disable wazuh-agent > /dev/null 2>&1
+        rm -f /etc/systemd/system/wazuh-agent.service
+        update-rc.d -f wazuh-agent remove
+        rm -f /etc/init.d/wazuh-agent
+        systemctl daemon-reload
```
Comment (alberpilot) [inline debs/SPECS/3.7.0/wazuh-agent/debian/postrm]: In comparison with the RPM package, we need to ensure the order of the commands in terms of functionality and homogeneity. Please review it and we can decide if first "daemon-reload" or "reset-failed"

File: rpms/SPECS/3.7.0/wazuh-agent-3.7.0.spec [hunk 2]
```diff
@@ -248,10 +248,13 @@ fi
 
 %preun
 
+# If the package is been uninstalled
 if [ $1 = 0 ]; then
-
+  # Stop the wazuh-manager
```
Comment (alberpilot) [inline rpms/SPECS/3.7.0/wazuh-agent-3.7.0.spec]: This comment has to be changed by wazuh-agent

**Comment by BraulioV** [comment]
Hi @alberpilot, 

with the last commit, I fixed all the things from your review, but, I found a bug in the .deb packages. A .deb package built with the changes from this branch, if you execute `apt install -y wazuh-manager && apt remove -y wazuh-manager && apt install wazuh-manager`, the package won't install the services.

I'm currently investigating why this is happening.

**Comment by BraulioV** [comment]
Hi @alberpilot,

after some research, the error is caused by deleting the services when running `apt remove wazuh-manager` or `apt remove wazuh-agent`. I will explain why.

When you execute `apt remove ....`, only the configuration files remains in the host. Those files are the ones listed in the `conffiles` file of the package, and all the files installed under `/etc`, so the files `/etc/init.d/wazuh-manager` and `/etc/systemd/system/wazuh-manager` are marked as configuration files too.

When you install the package after removing it, the package will try to restore the configuration files, but if we delete the services files when removing the package, those files aren't restored.

The solution is to only remove the files while running `apt purge` and let the files in the host when running `apt remove`. This will only solve the problem of `apt install wazuh-xxxxxx+ apt remove wazuh-xxxxxx + apt install wazuh-xxxxxx`, but the services will remain in the host.

To solve this, we can use `systemctl mask`, to mask the service and fully disable it when you run `apt remove`. This will mark the service and no one can execute it manually or automatically. Then, if you reinstalled the package again, the package will run `systemctl unmask` and enable the package. 

But, to include this fix, we must change the `.service` files for `systemd` from `/etc/systemd/system` to `/lib/systemd/system/wazuh-manager.service`. Otherwise, `systemctl maks` won't work.

Regards,
Braulio.

**Comment by alberpilot** [comment]
We are working this issue on this branch: 

https://github.com/wazuh/wazuh-packages/tree/3.7.1-improve-services

@BraulioV let's move on the mentioned branch. 
Thanks.

Best regards, 
Alberto R

**Pull request closed without merging**